diff --git a/Makefile.in b/Makefile.in
index 76fdd3c..67e8c35 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -29,6 +29,7 @@ INSTALL = @INSTALL@
 AR	= @AR@
 RANLIB	= @RANLIB@
 WINDRES	= @WINDRES@
+LIBS	= @LIBS@
 SDL_CFLAGS = @SDL_CFLAGS@
 SDL_LIBS = @SDL_LIBS@
 
@@ -72,13 +73,13 @@ $(objects):
 @PLAYMUS_DEPENDS@
 
 $(objects)/$(TARGET): $(OBJECTS) $(VERSION_OBJECTS)
-	$(LIBTOOL) --mode=link $(CC) -o $@ $(OBJECTS) $(VERSION_OBJECTS) $(LDFLAGS) $(EXTRA_LDFLAGS) $(LT_LDFLAGS)
+	$(LIBTOOL) --mode=link $(CC) -o $@ $(OBJECTS) $(VERSION_OBJECTS) $(LDFLAGS) $(LIBS) $(EXTRA_LDFLAGS) $(LT_LDFLAGS)
 
 $(objects)/playwave$(EXE): $(objects)/playwave.lo $(objects)/$(TARGET)
-	$(LIBTOOL) --mode=link $(CC) -o $@ $(objects)/playwave.lo $(SDL_CFLAGS) $(SDL_LIBS) $(objects)/$(TARGET)
+	$(LIBTOOL) --mode=link $(CC) -o $@ $(objects)/playwave.lo $(SDL_CFLAGS) $(SDL_LIBS) $(objects)/$(TARGET) $(LIBS)
 
 $(objects)/playmus$(EXE): $(objects)/playmus.lo $(objects)/$(TARGET)
-	$(LIBTOOL) --mode=link $(CC) -o $@ $(objects)/playmus.lo $(SDL_CFLAGS) $(SDL_LIBS) $(objects)/$(TARGET)
+	$(LIBTOOL) --mode=link $(CC) -o $@ $(objects)/playmus.lo $(SDL_CFLAGS) $(SDL_LIBS) $(objects)/$(TARGET) $(LIBS)
 
 install: all install-hdrs install-lib #install-bin
 install-hdrs:
diff --git a/configure.in b/configure.in
index 7d23387..87ba9e5 100644
--- a/configure.in
+++ b/configure.in
@@ -164,6 +164,10 @@ find_lib()
     done
 }
 
+dnl Check for libm
+AC_CHECK_LIBM
+LIBS="$LIBS $LIBM"
+
 dnl Check for SDL
 SDL_VERSION=1.2.10
 AM_PATH_SDL($SDL_VERSION,
