diff --git a/configure.in b/configure.in
index 87ba9e5..2f1a399 100644
--- a/configure.in
+++ b/configure.in
@@ -217,66 +217,12 @@ AC_ARG_ENABLE([music-mod-shared],
 AC_HELP_STRING([--enable-music-mod-shared], [dynamically load MOD support [[default=yes]]]),
               [], [enable_music_mod_shared=yes])
 if test x$enable_music_mod = xyes; then
-    have_libmikmod=no
-    libmikmod_maj=3
-    libmikmod_min=1
-    libmikmod_rev=10
-    libmikmod_ver="$libmikmod_maj.$libmikmod_min.$libmikmod_rev"
-    AC_PATH_PROG(LIBMIKMOD_CONFIG, libmikmod-config, no, [$PATH])
-    if test "$LIBMIKMOD_CONFIG" != "no" ; then
-        
-        CFLAGS_SAVED="$CFLAGS"
-        LIBS_SAVED="$LIBS"
-        CFLAGS="$CFLAGS -DLIBMIKMOD_MUSIC `$LIBMIKMOD_CONFIG --cflags`"
-        LIBS="$LIBS `$LIBMIKMOD_CONFIG --libs`"
-        have_libmikmod=yes
-        AC_MSG_CHECKING([for libmikmod - version >= $libmikmod_ver])
-        AC_TRY_RUN([
-#include "mikmod.h"
-#include "stdio.h"
-
-int main(int argc, char **argv)
-{
-long maj=$libmikmod_maj,min=$libmikmod_min,rev=$libmikmod_rev,ver=MikMod_GetVersion();
-/*printf("(%d.%d.%d) ",ver>>16,(ver>>8)&0xff,ver&0xff);*/
-if(ver>=((maj<<16)|(min<<8)|(rev))) {
-    printf("yes\n");
-    return 0;
-}
-printf("no\n*** libmikmod is older than %d.%d.%d, not using.\n",maj,min,rev);
-return 1;
-}
-], [], have_libmikmod=no; CFLAGS="$CFLAGS_SAVED"; LIBS="$LIBS_SAVED",
-        [echo $ac_n "cross compiling; assumed OK... $ac_c"])
-    fi
-
-    if test x$have_libmikmod = xyes; then
-        case "$host" in
-            *-*-darwin*)
-                mikmod_lib=[`find_lib libmikmod.dylib`]
-                ;;
-            *-*-cygwin* | *-*-mingw32*)
-                mikmod_lib=[`find_lib "mikmod*.dll"`]
-                ;;
-            *)
-                mikmod_lib=[`find_lib "libmikmod*.so.[0-9]"`]
-                if test x$mikmod_lib = x; then
-                    mikmod_lib=[`find_lib "libmikmod*.so.[0-9]*"`]
-                fi
-                ;;
-        esac
-        SOURCES="$SOURCES $srcdir/*_mod.c"
-        EXTRA_CFLAGS="$EXTRA_CFLAGS -DMOD_MUSIC `$LIBMIKMOD_CONFIG --cflags`"
-        if test x$enable_music_mod_shared = xyes && test x$mikmod_lib != x; then
-            echo "-- dynamic libmikmod -> $mikmod_lib"
-            EXTRA_CFLAGS="$EXTRA_CFLAGS -DMOD_DYNAMIC=\\\"$mikmod_lib\\\""
-        else
-            EXTRA_LDFLAGS="$EXTRA_LDFLAGS `$LIBMIKMOD_CONFIG --libs`"
-        fi
-    else
-        AC_MSG_WARN([*** Unable to find MikMod library (http://mikmod.raphnet.net/)])
-        AC_MSG_WARN([MOD support disabled])
-    fi
+	AM_PATH_LIBMIKMOD($libmikmod_ver,
+		CFLAGS="$CFLAGS -DLIBMIKMOD_MUSIC $LIBMIKMOD_CFLAGS"
+		SYSTEM_LIBS="$SYSTEM_LIBS $LIBMIKMOD_LIBS",
+		no_libmikmod=yes)
+else
+	no_libmikmod=yes
 fi
 
 AC_ARG_ENABLE([music-midi],
diff --git a/debian/patches/00_mikmod_syslibs.patch b/debian/patches/00_mikmod_syslibs.patch
index 794457b..e69de29 100644
--- a/debian/patches/00_mikmod_syslibs.patch
+++ b/debian/patches/00_mikmod_syslibs.patch
@@ -1,42 +0,0 @@
---- sdl-mixer1.2-1.2.6/configure.in.orig	2005-01-22 04:36:35.000000000 +0100
-+++ sdl-mixer1.2-1.2.6/configure.in	2005-01-22 04:37:27.000000000 +0100
-@@ -112,33 +112,12 @@
- [  --enable-music-libmikmod  enable MOD music via externl libmikmod [default=yes]],
-               , enable_music_libmikmod=yes)
- if test x$enable_music_libmikmod = xyes; then
--	AC_PATH_PROG(LIBMIKMOD_CONFIG, libmikmod-config, no, [$PATH])
--	if test "$LIBMIKMOD_CONFIG" = "no" ; then
--		no_libmikmod==yes
--	else
--		CFLAGS_SAVED="$CFLAGS"
--		LIBS_SAVED="$LIBS"
--		CFLAGS="$CFLAGS -DLIBMIKMOD_MUSIC `$LIBMIKMOD_CONFIG --cflags`"
--		LIBS="$LIBS `$LIBMIKMOD_CONFIG --libs`"
--		AC_MSG_CHECKING([for libmikmod - version >= $libmikmod_ver])
--		AC_TRY_RUN([
--#include "mikmod.h"
--#include "stdio.h"
--
--int main(int argc, char **argv)
--{
--	long maj=$libmikmod_maj,min=$libmikmod_min,rev=$libmikmod_rev,ver=MikMod_GetVersion();
--	/*printf("(%d.%d.%d) ",ver>>16,(ver>>8)&0xff,ver&0xff);*/
--	if(ver>=((maj<<16)|(min<<8)|(rev))) {
--		printf("yes\n");
--		return 0;
--	}
--	printf("no\n*** libmikmod is older than %d.%d.%d, not using.\n",maj,min,rev);
--	return 1;
--}
--],, no_libmikmod=yes;CFLAGS="$CFLAGS_SAVED";LIBS="$LIBS_SAVED",
--[echo $ac_n "cross compiling; assumed OK... $ac_c"])
--	fi
-+	AM_PATH_LIBMIKMOD($libmikmod_ver,
-+		CFLAGS="$CFLAGS -DLIBMIKMOD_MUSIC $LIBMIKMOD_CFLAGS"
-+		SYSTEM_LIBS="$SYSTEM_LIBS $LIBMIKMOD_LIBS",
-+		no_libmikmod=yes)
-+else
-+	no_libmikmod=yes
- fi
- 
- AC_ARG_ENABLE(music-mod,
