--- sdl-mixer1.2-1.2.6/configure.in.orig	2005-01-22 04:36:35.000000000 +0100
+++ sdl-mixer1.2-1.2.6/configure.in	2005-01-22 04:37:27.000000000 +0100
@@ -112,33 +112,12 @@
 [  --enable-music-libmikmod  enable MOD music via externl libmikmod [default=yes]],
               , enable_music_libmikmod=yes)
 if test x$enable_music_libmikmod = xyes; then
-	AC_PATH_PROG(LIBMIKMOD_CONFIG, libmikmod-config, no, [$PATH])
-	if test "$LIBMIKMOD_CONFIG" = "no" ; then
-		no_libmikmod==yes
-	else
-		CFLAGS_SAVED="$CFLAGS"
-		LIBS_SAVED="$LIBS"
-		CFLAGS="$CFLAGS -DLIBMIKMOD_MUSIC `$LIBMIKMOD_CONFIG --cflags`"
-		LIBS="$LIBS `$LIBMIKMOD_CONFIG --libs`"
-		AC_MSG_CHECKING([for libmikmod - version >= $libmikmod_ver])
-		AC_TRY_RUN([
-#include "mikmod.h"
-#include "stdio.h"
-
-int main(int argc, char **argv)
-{
-	long maj=$libmikmod_maj,min=$libmikmod_min,rev=$libmikmod_rev,ver=MikMod_GetVersion();
-	/*printf("(%d.%d.%d) ",ver>>16,(ver>>8)&0xff,ver&0xff);*/
-	if(ver>=((maj<<16)|(min<<8)|(rev))) {
-		printf("yes\n");
-		return 0;
-	}
-	printf("no\n*** libmikmod is older than %d.%d.%d, not using.\n",maj,min,rev);
-	return 1;
-}
-],, no_libmikmod=yes;CFLAGS="$CFLAGS_SAVED";LIBS="$LIBS_SAVED",
-[echo $ac_n "cross compiling; assumed OK... $ac_c"])
-	fi
+	AM_PATH_LIBMIKMOD($libmikmod_ver,
+		CFLAGS="$CFLAGS -DLIBMIKMOD_MUSIC $LIBMIKMOD_CFLAGS"
+		SYSTEM_LIBS="$SYSTEM_LIBS $LIBMIKMOD_LIBS",
+		no_libmikmod=yes)
+else
+	no_libmikmod=yes
 fi
 
 AC_ARG_ENABLE(music-mod,
