Index: sdl-mixer1.2-1.2.8/configure.in
===================================================================
--- sdl-mixer1.2-1.2.8.orig/configure.in	2008-05-21 11:51:07.000000000 +0000
+++ sdl-mixer1.2-1.2.8/configure.in	2008-05-21 11:51:18.000000000 +0000
@@ -317,7 +317,7 @@
                 echo "-- dynamic libvorbisfile -> $ogg_lib"
                 EXTRA_CFLAGS="$EXTRA_CFLAGS -DOGG_DYNAMIC=\\\"$ogg_lib\\\""
             else
-                EXTRA_LDFLAGS="$EXTRA_LDFLAGS -lvorbisfile -lvorbis -logg"
+                EXTRA_LDFLAGS="$EXTRA_LDFLAGS -lvorbisfile"
             fi
         fi
     fi
