diff --git a/music_cmd.c b/music_cmd.c
index 5a43ddc..e2dd1c5 100644
--- a/music_cmd.c
+++ b/music_cmd.c
@@ -54,10 +54,8 @@ MusicCMD *MusicCMD_LoadSong(const char *cmd, const char *file)
 		Mix_SetError("Out of memory");
 		return(NULL);
 	}
-	strncpy(music->file, file, (sizeof music->file)-1);
-	music->file[(sizeof music->file)-1] = '\0';
-	strncpy(music->cmd, cmd, (sizeof music->cmd)-1);
-	music->cmd[(sizeof music->cmd)-1] = '\0';
+	music->file = strdup(file);
+	music->cmd = strdup(cmd);
 	music->pid = 0;
 
 	/* We're done */
@@ -156,7 +154,7 @@ void MusicCMD_Start(MusicCMD *music)
 
 	    /* Child process - executes here */
 	    case 0: {
-		    char command[PATH_MAX];
+		    char *command;
 		    char **argv;
 
 		    /* Unblock signals in case we're called from a thread */
@@ -167,11 +165,12 @@ void MusicCMD_Start(MusicCMD *music)
 		    }
 
 		    /* Execute the command */
-		    strcpy(command, music->cmd);
+		    command = strdup(music->cmd);
 		    argv = parse_args(command, music->file);
 		    if ( argv != NULL ) {
 			execvp(argv[0], argv);
 		    }
+		    free(command);
 
 		    /* exec() failed */
 		    perror(argv[0]);
@@ -220,6 +219,8 @@ void MusicCMD_Resume(MusicCMD *music)
 /* Close the given music stream */
 void MusicCMD_FreeSong(MusicCMD *music)
 {
+	free(music->file);
+	free(music->cmd);
 	free(music);
 }
 
diff --git a/music_cmd.h b/music_cmd.h
index 41757a8..5f82cde 100644
--- a/music_cmd.h
+++ b/music_cmd.h
@@ -31,8 +31,8 @@
 # include <linux/limits.h>
 #endif
 typedef struct {
-	char file[PATH_MAX];
-	char cmd[PATH_MAX];
+	char *file;
+	char *cmd;
 	pid_t pid;
 } MusicCMD;
 
