diff --git a/timidity/config.h b/timidity/config.h
index ace589b..2478c09 100644
--- a/timidity/config.h
+++ b/timidity/config.h
@@ -173,6 +173,7 @@ typedef char int8;
    then specify the library directory in the configuration file. */
 #define CONFIG_FILE	"timidity.cfg"
 #define CONFIG_FILE_ETC "/etc/timidity.cfg"
+#define CONFIG_FILE_ETC_TIMIDITY_FREEPATS "/etc/timidity/freepats.cfg"
 
 #if defined(__WIN32__) || defined(__OS2__)
 #define DEFAULT_PATH	"C:\\TIMIDITY"
diff --git a/timidity/timidity.c b/timidity/timidity.c
index 1dea30c..9898562 100644
--- a/timidity/timidity.c
+++ b/timidity/timidity.c
@@ -298,7 +298,9 @@ int Timidity_Init(int rate, int format, int channels, int samples)
   if (!env || read_config_file(env)<0) {
     if (read_config_file(CONFIG_FILE)<0) {
       if (read_config_file(CONFIG_FILE_ETC)<0) {
-        return(-1);
+        if (read_config_file(CONFIG_FILE_ETC_TIMIDITY_FREEPATS)<0) {
+          return(-1);
+        }
       }
     }
   }
